instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/instrumentation.rb

```diff
@@ -64,12 +64,13 @@ def gem_version
 
         def require_dependencies
           require_relative 'patches/base'
-          require_relative 'patches/active_job_callbacks'
+          require_relative 'handlers'
         end
 
         def patch_activejob
-          ::ActiveJob::Base.prepend(Patches::Base)
-          ::ActiveJob::Base.prepend(Patches::ActiveJobCallbacks)
+          ::ActiveJob::Base.prepend(Patches::Base) unless ::ActiveJob::Base.ancestors.include?(Patches::Base)
```

Comment:
I assume this is for activejob < 6.1? Is this still required for rails 6.1? Will there be double span if on activejob 6.0 (since it both prepend and subscribe)

Reply:
This is required for all versions of `ActiveJob` unfortunately and only introduces patches for context propagation. It no longer contains any code that creates spans. 

ActiveJob does not have a concept of "headers" that we can use to propagate the context so we have to add a custom serializer that propagates the trace context. 

Thinking about this a bit, I am not sure why we need this check here. 🤔 Did I do this only for unit tests?

Reply:
ah, that makes sense, thanks. I thought require base will also include `active_job_callbacks.rb`, but now it looks like it doesn't. Do we still need `active_job_callbacks.rb`?

> Did I do this only for unit tests?

Yes, it seems like without unless, the test will have method redefined
```
/app/instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/patches/base.rb:15: warning: method redefined; discarding old __otel_headers=
```

Reply:
I see. Did I not delete the old callbacks file?